### PR TITLE
getState enforcePeriodicBox with triclinic unit cells

### PR DESCRIPTION
**Pull request by rmcgibbo**
cc #1229.

### CODE REVIEW

File: tests/TestEnforcePeriodicBox.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+#include "openmm/NonbondedForce.h"
+#include "openmm/Platform.h"
+#include "openmm/VerletIntegrator.h"
+#include "sfmt/SFMT.h"
+#include <iostream>
+
+using namespace OpenMM;
+using namespace std;
+
+void testTrunactedOctahedron() {
```
Comment (swails) [inline tests/TestEnforcePeriodicBox.cpp]: Truncated

**Comment by swails** [comment]
This fixes the problem, btw.  Thanks!

**Comment by rmcgibbo** [comment]
I think it would be still worthwhile to think about how to make the optimizer more robust, but I guess that requires a different test case.

**Comment by swails** [comment]
Here's the error in one of the Travis builds: https://travis-ci.org/pandegroup/openmm/jobs/88685253#L1291-L1293

I'm pretty sure this is a race condition due to some dependencies that are not correctly set.  Here's what I've observed on my own machines:

- `make install` always works.
- `make -j# install` sometimes fails on one thread with an error like that one, and that failure brings down the whole install.  However, subsequent `make install` or `make -j# install` commands always complete just fine.

So we can do one of two things here:

1. Hunt down where the missing dependency is coming from and fix it
2. Change `make -j2 install` to `make -j2 install || make -j2 install`

The latter is easier and should be just as robust.

**Comment by rmcgibbo** [comment]
No, the error is fixed by #1232.

**Comment by swails** [comment]
I've been getting the strange parallel-make errors basically since I started compiling OpenMM.  Will #1232 fix those?

**Comment by rmcgibbo** [comment]
That error applies regardless of things being in parallel. There's an extra newline in the `CMAKE_OSX_SYSROOT`, which just kills the whole construction of the input arguments to the compiler, I think. Because the variables are cached in CMake, you have to recreate the build directory to see it though.

**Comment by rmcgibbo** [comment]
It seems like all of the RPMD tests are failing.

**Comment by rmcgibbo** [comment]
perhaps https://github.com/pandegroup/openmm/blob/master/plugins/rpmd/openmmapi/src/RPMDIntegrator.cpp#L132-L145 also needs to be updated.

**Comment by peastman** [comment]
Exactly what I was just about to post.

**Comment by rmcgibbo** [comment]
Okay, let's see if that fixes it.

**Comment by peastman** [comment]
Looks good!  The one problem was TestOpenCLRPMDDouble, which timed out.  Possibly that test takes too long to run on travis?

**Comment by rmcgibbo** [comment]
Yeah, it timed out the first time it was run, which is expected since we need to enforce the time limit on ctest, but then it was rerun and timed out again with `No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.` It makes sense that it would be one of the slower tests.

**Pull request merged**